scalding-core/src/main/scala/com/twitter/scalding/ArgHelp.scala

```diff
@@ -0,0 +1,108 @@
+package com.twitter.scalding
+
+sealed trait DescribedArg {
+  def key: String
+  def description: String
+}
+
+case class RequiredArg(key: String, description: String) extends DescribedArg
+case class OptionalArg(key: String, description: String) extends DescribedArg
+case class ListArg(key: String, description: String) extends DescribedArg
+case class BooleanArg(key: String, description: String) extends DescribedArg
+
+class HelpException extends RuntimeException("User asked for help")
+class DescriptionValidationException(msg: String) extends RuntimeException(msg)
+
+trait ArgHelper {
+  /**
+   * Similar to describe but validate all args are described
+   * @param describedArgs List of Argument Descriptions
+   * @param ex Input Execution
+   * @return Output Execution
+   */
+  def validatedDescribe[T](describedArgs: Seq[DescribedArg], ex: Execution[T]): Execution[T] = {
+    val describedKeys = describedArgs.map(_.key).toSet
+    Execution.getArgs.flatMap { args =>
+      val missingKeys = args.m.keySet.filter(_.nonEmpty).diff(describedKeys)
+
+      if (args.boolean("help")) {
+        helpRequest(describedArgs)
+      } else {
+        if (missingKeys.nonEmpty) {
+          val msg = missingKeys.mkString(", ")
+          throw new DescriptionValidationException(s"Must describe missing keys : $msg")
+        } else {
+          ex
+        }
+      }
+    }
+  }
+
+  /**
+   * Describe the Arguments of this Execution.  By running --help the args will output
+   * and the execution will end
+   *
+   * @param describedArgs List of Argument Descriptions
+   * @param ex Input Execution
+   * @return Output Execution
+   */
+  def describe[T](describedArgs: Seq[DescribedArg], ex: Execution[T]): Execution[T] = {
+    Execution.getArgs.flatMap { args =>
+      if (args.boolean("help")) {
+        helpRequest(describedArgs)
+      } else {
+        ex
+      }
+    }
+  }
+
+  def helpRequest[T](describedArgs: Seq[DescribedArg]): Execution[T] = {
```

Comment:
I think what he means is just a function signature like:
```
def helpRequest(describedArgs: Seq[DescribedArg]): Nothing = {
```

Since you never actually return an Execution